src/Refactoring-Core/RBAddClassRefactoring.class.st

```diff
@@ -77,17 +119,28 @@ RBAddClassRefactoring >> storeOn: aStream [
 	superclass storeOn: aStream.
 	aStream nextPutAll: ' subclasses: '.
 	subclasses asArray storeOn: aStream.
+	aStream nextPutAll: ' traits: '.
+	traits asArray storeOn: aStream.
 	aStream nextPutAll: ' category: '.
 	category storeOn: aStream.
 	aStream nextPut: $)
 ]
 
 { #category : #transforming }
 RBAddClassRefactoring >> transform [
-	(self model)
-		defineClass: ('<1p> subclass: #<2s> instanceVariableNames: '''' classVariableNames: '''' poolDictionaries: '''' category: <3p>' 
-					expandMacrosWith: superclass
-					with: className
-					with: category asString);
-		reparentClasses: subclasses to: (self model classNamed: className asSymbol)
+
+	self model
+		defineClass:
+			('<1p> subclass: #<2s><3s> instanceVariableNames: '''' classVariableNames: '''' poolDictionaries: '''' category: <4p>'
+				 expandMacrosWith: superclass
+				 with: className
+				 with: (String streamContents: [ :stream | 
+							  traits ifNotEmpty: [ 
+									  stream nextPutAll: ' uses: '.
+									  traits
+										  do: [ :each | stream nextPutAll: each name ]
+										  separatedBy: [ stream nextPutAll: ' + ' ] ] ])
```

Comment:
probably this code handles only simple case, use a full traits but do not handle traits composition.